packages/Symfony/src/Rector/Console/ConsoleExecuteReturnIntRector.php

```diff
@@ -0,0 +1,125 @@
+<?php declare(strict_types=1);
+
+namespace Rector\Symfony\Rector\Console;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\FunctionLike;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Scalar\LNumber;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Return_;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+/**
+ * Covers:
+ * - https://github.com/symfony/symfony/pull/33775/files
+ * @see \Rector\Symfony\Tests\Rector\Console\ConsoleExecuteReturnIntRector\ConsoleExecuteReturnIntRectorTest
+ */
+final class ConsoleExecuteReturnIntRector extends AbstractRector
+{
+    private const COMMAND_CLASS = '\Symfony\Component\Console\Command\Command';
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Returns int from Command::execute command', [
+            new CodeSample(
+                <<<'PHP'
+class SomeCommand extends Command
+{
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        return null;
+    }
+}
+PHP
+                ,
+                <<<'PHP'
+class SomeCommand extends Command
+{
+    public function index(InputInterface $input, OutputInterface $output): int
+    {
+        return 0;
+    }
+}
+PHP
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->getName($node) !== 'execute') {
+            return null;
+        }
+
+        $class = $node->getAttribute(AttributeKey::CLASS_NODE);
+        if (! $class || ! $class instanceof Class_) {
+            return null;
+        }
+
+        if (!$class->extends || $class->extends->toCodeString() !== self::COMMAND_CLASS) {
+            return null;
+        }
+
+        $node->returnType = new Identifier('int');
+
+        return $this->addReturn0ToMethod($node);
+    }
+
+    private function setReturnTo0InsteadOfNull(Return_ $return)
+    {
+        if (! $return->expr ) {
+            $return->expr = new LNumber(0);
+        }
+
+        if ($return->expr instanceof ConstFetch && $this->getName($return->expr) === 'null') {
+            $return->expr = new LNumber(0);
+        }
+
+        if ($return->expr instanceof MethodCall || $return->expr instanceof StaticCall) {
+            // how to get the Node of teh called method?!
```

Comment:
@TomasVotruba i guess i need help here, am i able to the get the nodes that are called?!

or do you see another way to handle the case from the `return-function-call` testcase?

Reply:
From top of my head:

```php
$returnStaticType = $this->getStaticType($return->expr);
if ($returnStaticType instanceof IntegerType) {
    return null;
}

$return->expr = new LNumber(0);
```

Reply:
I got that covered, what I have trouble with is returning the value of a method call, e.g.
```
return $this->doRun();
```
My first idea was to apply the rector to the `doRun()` method too, but I have no idea how to get the node of the called method.

Reply:
Good!

What happens when you try the code I wrote above?

Reply:
That leads to:
```diff
    public function execute(InputInterface $input, OutputInterface $output)
    {
-       return $this->doSomething();
+      return 0;
    }

    private function doSomething()
    {
        return null;
    }
```

but this is want i want to accomplish:
```diff
    public function execute(InputInterface $input, OutputInterface $output)
    {
        return $this->doSomething();
    }

    private function doSomething()
    {
-       return null;
+      return 0;
    }
```

I need to be able to somehow get the node of the called function, in this example the `doSomething()` method, but i don't know how as it seems that the `MethodCall`-Node just has a name of the called method but no reference to the node of that method.

Reply:
nevermind i got now, guess i just needed a bit of rubber ducking :laughing: